drivers/power/max17851/max17851.c

```diff
@@ -0,0 +1,748 @@
+/***************************************************************************//**
+ *   @file   max17851.c
+ *   @brief  Source file of the MAX17851 driver
+ *   @author Radu Sabau (radu.sabau@analog.com)
+********************************************************************************
+ * Copyright 2025(c) Analog Devices, Inc.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived from this
+ *    software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES, INC. “AS IS” AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO
+ * EVENT SHALL ANALOG DEVICES, INC. BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+ * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#include "max17851.h"
+#include "no_os_alloc.h"
+#include "no_os_error.h"
+
+/**
+ * @brief Read any UART related errors in case of failures communication wise.
+ * @param desc - Instance of UART
+ * @return Status byte that contains any UART errors that appeared.
+ */
+static uint32_t max17851_uart_get_errors(struct no_os_uart_desc *desc)
+{
+	struct max17851_desc *extra;
+	uint8_t reg_val;
+	int ret;
+
+	extra = desc->extra;
+	ret = max17851_reg_read(extra, MAX17851_STATUS_GEN_REG, &reg_val);
+	if (ret)
+		return ret;
+
+	return (int32_t)reg_val;
```

Comment:
same here, you explicitly cast on signed value, but the return type is unsigned.